extensions/roc-package-web-app-react/src/app/server/reactRenderer.js

```diff
@@ -30,18 +31,42 @@ const whiteListed = () => (
 
 const appConfig = whiteListed();
 
-export function initRenderPage({ script, css }, distMode, devMode, Header) {
+function setupTemplate(devMode) {
     const templatePaths = [].concat(
         // Combine paths from highest priority to lowest
         rocConfig.runtime.template.path || [],
         invokeHook('get-template-paths'),
         defaultTemplatePath
     ).map(path => getAbsolutePath(path));
 
-    nunjucks.configure(templatePaths, {
+    let mainTemplate = rocConfig.runtime.template.name;
+    const baseTemplate = 'roc-package-web-app-react/main.njk';
+    const inheritance = {};
+    invokeHook('inherit-template')(({ namespace, template }) => {
```

Comment:
Should we manage that an action might not return values for `namespace` and `template`? Maybe throw an exception if that happens since it otherwise might break the chain?